lib/determinator/control.rb

```diff
@@ -115,12 +115,41 @@ def choose_target_group(feature, properties)
 
         tg.constraints.reduce(true) do |fit, (scope, *required)|
           present = [*normalised_properties[scope]]
-          fit && (required.flatten & present.flatten).any?
+          fit && (has_any(required, present) || has_any_app_version?(scope, required, present))
         end
       # Must choose target group deterministically, if more than one match
       }.sort_by { |tg| tg.rollout }.last
     end
 
+    def has_any(required, present)
+      (required.flatten & present.flatten).any?
+    end
+
+    def has_any_app_version?(scope, required, present)
+      return false unless scope == 'app_version'
+
+      required_constraints = required.flatten.map {|constraint|
+        begin
+          operator, version = constraint.match(/([<>]=?)(.*)/).captures
+          {:operator => operator, :gem_version => Gem::Version.new(version)}
+        rescue
+          # Ignore malformed constraints
+          next
+        end
+      }.compact
+      present.flatten.select {|version|
+        begin
+          present_gem_version = Gem::Version.new(version)
+        rescue
+          # Ignore malformed app_version value
+          next false
+        end
+        required_constraints.all? {|constraint|
+          present_gem_version.method(constraint[:operator]).(constraint[:gem_version])
```

Comment:
I'd strongly suggest using a gem for this, perhaps https://www.rubydoc.info/gems/semantic/1.6.1

If you want to stick to this approach the idomatic ruby way to would be to do something like 

`present_gem_version.public_send(constraint[:operator], constraint[:gem_version])`

rather than what you're doing here - getting a handle to the actual method.

Reply:
Oh! I was looking for that and ended up thinking it didn't exist. I'll have a look into this. Thanks!